### PR TITLE
3 scoring subsystem skeleton

### LINKED ISSUE #3: Scoring Subsystem Skeleton
- [x] Create a subsystem named `Scoring.java` and make it extend PIDSubsystem.
- [x] Create two `CanSparkMAX`, one named `leftScoringElevatorMotor` and the other named `rightScoringElevatorMotor`
- [x] create an absolute encoder for the left motor.
- [x] in the constructor inherit a PIDController (values for which should be stored in a constants file).
- [x] Create another `CanSparkMAX`, named `rollerMotor`.
- [x] Make the right motor mirror the left one.
- [x] Create a method that runs the roller motor and a method to stop it.
- [x] Create an enum for Scoring elevator setpoints. 
- [x] Three setpoints: ground, amp, and trap. 
- [x] Create the @Override methods for a PIDSubsystem; void useOutput, double getMeasurements, void setSetpoint.
- [x] Create a getPosition method that returns the position of the encoder.
- [x] setP, setI, setD methods
- [x] steal from Wrist.java and Shoulder.java
-

### PR DESCRIPTION
**Pull request by Griffin9881**
### Pull Request

---------------------------------------------

#### Issue Number

Closes #3 

#### Comments

finished with skeleton for scoring subsystem

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Scoring.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    protected void useOutput(double output, double setpoint) {
+
+    }
+
+    @Override
+    protected double getMeasurement() {
+        return getPosition();
+    }
+
+    public void setClampSetpoint(double setpoint) {
```
Comment (matunkar) [inline src/main/java/frc/robot/subsystems/Scoring.java:71]:
```suggestion
    public void clampSetpoint(double setpoint) {
```

**Review by matunkar — APPROVED** [review]
looks good, should code be added to useOutput?

File: src/main/java/frc/robot/subsystems/Scoring.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (instance == null) {
+            instance = new Scoring();
+        }
+        return instance;
+    }
+
+    public Scoring() {
+        super(new PIDController(ScoringConstants.kP, ScoringConstants.kI, ScoringConstants.kD));
+        rightScoringElevatorMotor.restoreFactoryDefaults();
```
Comment (reediculous456) [inline src/main/java/frc/robot/subsystems/Scoring.java:52]: need to restore defaults and burn flash of the roller motor

File: src/main/java/frc/robot/subsystems/Scoring.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        rightScoringElevatorMotor.setIdleMode(IdleMode.kCoast);
+        leftScoringElevatorMotor.setIdleMode(IdleMode.kCoast);
+        rightScoringElevatorMotor.follow(leftScoringElevatorMotor);
+        rightScoringElevatorMotor.burnFlash();
+        leftScoringElevatorMotor.burnFlash();
+    }
+
+    @Override
+    protected void useOutput(double output, double setpoint) {
+
```
Comment (reediculous456) [inline src/main/java/frc/robot/subsystems/Scoring.java:63]: this function needs an implementation

File: src/main/java/frc/robot/subsystems/Scoring.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    public void setSetpoint(ElevatorSetpoints setpoint) {
+        setSetpoint(setpoint.value);
+    }
+
+    public double getPosition() {
+        return leftScoringElevatorEncoder.getPosition();
+    }
+
+    public void runRoller(double speed) {
+        rollerMotor.set(speed);
```
Comment (reediculous456) [inline src/main/java/frc/robot/subsystems/Scoring.java:85]: This speed needs clamped

**Review by reediculous456 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/constants/ScoringConstants.java [hunk 5]
```diff
@@ -10,6 +10,8 @@ public class ScoringConstants {
     public static final double kTrapSetpoint = 0;
 
     public static final double kMaxDistance = 0;
+    public static final double kMaxSpeed = 0;
+    public static final double kMaxPower = 0;
```
Comment (reediculous456) [inline src/main/java/frc/robot/constants/ScoringConstants.java:14]:
```suggestion
    public static final double kMaxSpeed = 0;
    public static final double kMinSpeed = 0;
```

**Review by reediculous456 — CHANGES_REQUESTED** [review]

**Review by reediculous456 — APPROVED** [review]

**Pull request merged**